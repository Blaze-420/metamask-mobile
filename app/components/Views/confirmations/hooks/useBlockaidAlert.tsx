import React from 'react';
import { Text } from 'react-native';

import Alert from '../types/confirm';

const useBlockaidAlert = (): Alert | undefined => ({
  alertDetails: ['Alert details - 1', 'Alert details - 2'],
  key: 'blockaid',
  component: <Text>report url: www.google.com</Text>,
  isBlocking: true,
  title: 'blockaid alert',
  message: 'Blockaid has reported this request tobe malicious',
  severity: 'danger',
});

export default useBlockaidAlert;
