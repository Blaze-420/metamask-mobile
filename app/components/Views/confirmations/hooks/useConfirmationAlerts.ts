import { useMemo } from 'react';
import { Alert } from '../types/confirm';
import useConfirmationOriginAlerts from './alerts/useConfirmationOriginAlerts';
import useBlockaidAlerts from './alerts/useBlockaidAlerts';
import useDomainMismatchAlerts from './alerts/signatures/useDomainMismatchAlerts';

function useSignatureAlerts(): Alert[] {
  const domainMismatchAlerts = useDomainMismatchAlerts();

  return useMemo(
    () => [...domainMismatchAlerts],
    [ domainMismatchAlerts],
  );
}

export default function useConfirmationAlerts(): Alert[] {
  const blockaidAlerts = useBlockaidAlerts();
  const confirmationOriginAlerts = useConfirmationOriginAlerts();
  const signatureAlerts = useSignatureAlerts();

  return useMemo(
    () => [
      ...blockaidAlerts,
      ...confirmationOriginAlerts,
      ...signatureAlerts,
    ],
    [
      blockaidAlerts,
      confirmationOriginAlerts,
      signatureAlerts,
    ],
  );
}
