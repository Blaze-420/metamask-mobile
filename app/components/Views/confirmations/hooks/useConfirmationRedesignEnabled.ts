import { useMemo } from 'react';
import { useSelector } from 'react-redux';
import {
  TransactionMeta,
  TransactionType,
} from '@metamask/transaction-controller';
import { ApprovalType } from '@metamask/controller-utils';

import { isHardwareAccount } from '../../../../util/address';
import { selectRemoteFeatureFlags } from '../../../../selectors/featureFlagController';
import { useTransactionMetadata } from './useTransactionMetadata';
import useApprovalRequest from './useApprovalRequest';

type RedesignRequest = {
  confirmation_redesign: ConfirmationRedesignRemoteFlags;
  approvalRequestType?: ApprovalType;
  fromAddress: string;
  transactionMetadata?: TransactionMeta;
};

type ConfirmationRedesignRemoteFlags = {
  signatures: boolean;
  stakingDeposit: boolean;
};

const REDESIGNED_SIGNATURE_TYPES = [
  ApprovalType.EthSignTypedData,
  ApprovalType.PersonalSign,
];

const REDESIGNED_TRANSACTION_TYPES = [TransactionType.stakingDeposit];

function isRedesignedSignature({
  confirmation_redesign,
  fromAddress,
  approvalRequestType,
}: RedesignRequest) {
  return (
    confirmation_redesign?.signatures &&
    // following condition will ensure that user is redirected to old designs for hardware wallets
    !isHardwareAccount(fromAddress) &&
    approvalRequestType &&
    REDESIGNED_SIGNATURE_TYPES.includes(approvalRequestType as ApprovalType)
  );
}

function isRedesignedStaking({
  approvalRequestType,
  confirmation_redesign,
  transactionMetadata,
}: RedesignRequest) {
  if (
    !transactionMetadata ||
    approvalRequestType !== ApprovalType.Transaction
  ) {
    return false;
  }

  return (
    confirmation_redesign?.stakingDeposit &&
    REDESIGNED_TRANSACTION_TYPES.includes(
      transactionMetadata.type as TransactionType,
    )
  );
}

export const useConfirmationRedesignEnabled = () => {
  const { approvalRequest } = useApprovalRequest();
  const transactionMetadata = useTransactionMetadata();
  const { confirmation_redesign } = useSelector(selectRemoteFeatureFlags) as {
    confirmation_redesign: ConfirmationRedesignRemoteFlags;
  };

  const approvalRequestType = approvalRequest?.type as ApprovalType;
  const fromAddress = approvalRequest?.requestData?.from;

  const redesignRequest: RedesignRequest = {
    approvalRequestType,
    confirmation_redesign,
    fromAddress,
    transactionMetadata,
  };

  const isRedesignedEnabled = useMemo(
    () =>
      isRedesignedSignature(redesignRequest) ||
      isRedesignedStaking(redesignRequest),
    [
      approvalRequestType,
      confirmation_redesign,
      fromAddress,
      transactionMetadata,
    ],
  );

  return { isRedesignedEnabled };
};
