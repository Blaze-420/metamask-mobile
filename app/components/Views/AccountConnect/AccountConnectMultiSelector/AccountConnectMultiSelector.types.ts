// Third party dependencies.
import { ImageSourcePropType } from 'react-native';

/**
 * Enum to track states of the account connect multi selector screen.
 */
export enum AccountConnectMultiSelectorScreens {
  AccountMultiSelector = 'AccountMultiSelector',
  AddAccountActions = 'AddAccountActions',
}

// External dependencies.
import { ConnectionProps } from '../../../../core/SDKConnect/Connection';
import { UseAccounts } from '../../../hooks/useAccounts';
import { IconName } from '../../../../component-library/components/Icons/Icon';
import { USER_INTENT } from '../../../../constants/permissions';
import { CaipAccountId } from '@metamask/utils';

/**
 * AccountConnectMultiSelector props.
 */
export interface AccountConnectMultiSelectorProps
  extends Omit<UseAccounts, 'evmAccounts'> {
  defaultSelectedAddresses: string[];
  onSubmit: (addresses: string[]) => void;
  isLoading?: boolean;
  hostname: string;
  isAutoScrollEnabled?: boolean;
  onBack: () => void;
  connection?: ConnectionProps;
  screenTitle?: string;
  isRenderedAsBottomSheet?: boolean;
  showDisconnectAllButton?: boolean;
}
