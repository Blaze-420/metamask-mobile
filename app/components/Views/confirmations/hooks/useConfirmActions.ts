import { useCallback } from 'react';

import PPOMUtil from '../../../../lib/ppom/ppom-util';
import { MetaMetricsEvents } from '../../../hooks/useMetrics';
import { isSignatureRequest } from '../utils/confirm';
import { useQRHardwareContext } from '../context/QRHardwareContext/QRHardwareContext';
import useApprovalRequest from './useApprovalRequest';
import { useLedgerWallet } from './useLedgerWallet';
import { useSignatureMetrics } from './useSignatureMetrics';

export const useConfirmActions = () => {
  const {
    onConfirm: onRequestConfirm,
    onReject: onRequestReject,
    approvalRequest,
  } = useApprovalRequest();
  const { captureSignatureMetrics } = useSignatureMetrics();
  const {
    cancelQRScanRequestIfPresent,
    isQRSigningInProgress,
    setScannerVisible,
  } = useQRHardwareContext();
  const { isLedgerAccount, openLedgerSignModal } = useLedgerWallet();

  const isSignatureReq =
    approvalRequest?.type && isSignatureRequest(approvalRequest?.type);

  const onReject = useCallback(async () => {
    // eslint-disable-next-line
    console.log('=================== INTO REJECT ===============');
    await cancelQRScanRequestIfPresent();
    onRequestReject();
    if (isSignatureReq) {
      captureSignatureMetrics(MetaMetricsEvents.SIGNATURE_REJECTED);
      PPOMUtil.clearSignatureSecurityAlertResponse();
    }
  }, [
    cancelQRScanRequestIfPresent,
    captureSignatureMetrics,
    onRequestReject,
    isSignatureReq,
  ]);

  const onApprovalConfirm = useCallback(async () => {
    // eslint-disable-next-line
    console.log('=================== INTO CONFIRM ===============');
    await onRequestConfirm({
      waitForResult: true,
      deleteAfterResult: true,
      handleErrors: false,
    });
    if (isSignatureReq) {
      captureSignatureMetrics(MetaMetricsEvents.SIGNATURE_APPROVED);
      PPOMUtil.clearSignatureSecurityAlertResponse();
    }
  }, [captureSignatureMetrics, onRequestConfirm, isSignatureReq]);

  const onConfirm = useCallback(async () => {
    if (isLedgerAccount) {
      // eslint-disable-next-line
      console.log('=================== LEDGER CONFIRM ===============');
      await openLedgerSignModal(onApprovalConfirm, onReject);
      return;
    }
    if (isQRSigningInProgress) {
      setScannerVisible(true);
      return;
    }
    await onApprovalConfirm();
  }, [
    isLedgerAccount,
    isQRSigningInProgress,
    onApprovalConfirm,
    openLedgerSignModal,
    setScannerVisible,
    onReject,
  ]);

  return { onConfirm, onReject };
};
