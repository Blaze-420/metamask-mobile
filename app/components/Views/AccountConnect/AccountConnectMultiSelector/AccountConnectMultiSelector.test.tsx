import React from 'react';
import { ImageSourcePropType } from 'react-native';
import { fireEvent } from '@testing-library/react-native';
import renderWithProvider from '../../../../util/test/renderWithProvider';
import AccountConnectMultiSelector from './AccountConnectMultiSelector';
import { backgroundState } from '../../../../util/test/initial-root-state';
import Engine from '../../../../core/Engine';
import { ConnectedAccountsSelectorsIDs } from '../../../../../e2e/selectors/Browser/ConnectedAccountModal.selectors';
import { AccountListBottomSheetSelectorsIDs } from '../../../../../e2e/selectors/wallet/AccountListBottomSheet.selectors';
import { ConnectAccountBottomSheetSelectorsIDs } from '../../../../../e2e/selectors/Browser/ConnectAccountBottomSheet.selectors';
import { IconName } from '../../../../component-library/components/Icons/Icon';
import { Account } from '../../../../components/hooks/useAccounts';
import { KeyringTypes } from '@metamask/keyring-controller';

const mockNavigate = jest.fn();
const mockOnSelectAddress = jest.fn();
const mockOnUserAction = jest.fn();
const mockOnBack = jest.fn();
const mockOnPrimaryActionButtonPress = jest.fn();

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: mockNavigate,
  }),
}));

jest.mock('../../../../core/Engine', () => ({
  context: {
    PermissionController: {
      revokeAllPermissions: jest.fn(),
    },
    AccountsController: {
      state: {
        internalAccounts: {
          accounts: {
            '0x1234': {
              address: '0x1234',
              name: 'Account 1',
              type: 'simple',
            },
            '0x5678': {
              address: '0x5678',
              name: 'Account 2',
              type: 'simple',
            },
          },
        },
      },
    },
  },
}));

const mockAccounts = [
  {
    address: '0x1234',
    name: 'Account 1',
    balance: '0x1',
    type: KeyringTypes.simple,
    yOffset: 0,
    isSelected: false,
  },
  {
    address: '0x5678',
    name: 'Account 2',
    balance: '0x2',
    type: KeyringTypes.simple,
    yOffset: 0,
    isSelected: false,
  },
];

const mockEnsByAccountAddress = {
  '0x1234': 'test1.eth',
  '0x5678': 'test2.eth',
};

const defaultProps = {
  accounts: mockAccounts,
  ensByAccountAddress: mockEnsByAccountAddress,
  selectedAddresses: ['0x1234'],
  onSelectAddress: mockOnSelectAddress,
  isLoading: false,
  onUserAction: mockOnUserAction,
  urlWithProtocol: 'https://test.com',
  hostname: 'test.com',
  onBack: mockOnBack,
  isRenderedAsBottomSheet: true,
  showDisconnectAllButton: true,
  favicon: { uri: 'https://test.com/favicon.ico' } as ImageSourcePropType,
  secureIcon: 'LockIcon' as IconName,
};

describe('AccountConnectMultiSelector', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { toJSON } = renderWithProvider(
      <AccountConnectMultiSelector {...defaultProps} />,
      { state: { engine: { backgroundState } } },
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('displays accounts list', () => {
    const { getByTestId } = renderWithProvider(
      <AccountConnectMultiSelector {...defaultProps} />,
      { state: { engine: { backgroundState } } },
    );

    expect(
      getByTestId(AccountListBottomSheetSelectorsIDs.ACCOUNT_LIST_ID),
    ).toBeDefined();
  });

  xit('handles select all checkbox correctly', () => {
    const { getByText } = renderWithProvider(
      <AccountConnectMultiSelector {...defaultProps} />,
      { state: { engine: { backgroundState } } },
    );

    const selectAllCheckbox = getByText('accounts.select_all');
    fireEvent.press(selectAllCheckbox);

    expect(mockOnSelectAddress).toHaveBeenCalledWith(['0x1234', '0x5678']);
  });

  xit('handles unselect all correctly', () => {
    const { getByText } = renderWithProvider(
      <AccountConnectMultiSelector
        {...defaultProps}
        selectedAddresses={['0x1234', '0x5678']}
      />,
      { state: { engine: { backgroundState } } },
    );

    const selectAllCheckbox = getByText('accounts.select_all');
    fireEvent.press(selectAllCheckbox);

    expect(mockOnSelectAddress).toHaveBeenCalledWith([]);
  });

  it('shows update button when accounts are selected', () => {
    const { getByTestId } = renderWithProvider(
      <AccountConnectMultiSelector {...defaultProps} />,
      { state: { engine: { backgroundState } } },
    );

    expect(
      getByTestId(ConnectAccountBottomSheetSelectorsIDs.SELECT_MULTI_BUTTON),
    ).toBeDefined();
  });

  xit('shows disconnect button when no accounts are selected', () => {
    const { getByTestId } = renderWithProvider(
      <AccountConnectMultiSelector {...defaultProps} selectedAddresses={[]} />,
      { state: { engine: { backgroundState } } },
    );

    expect(getByTestId(ConnectedAccountsSelectorsIDs.DISCONNECT)).toBeDefined();
  });

  xit('handles disconnect all action', () => {
    const { getByTestId } = renderWithProvider(
      <AccountConnectMultiSelector {...defaultProps} selectedAddresses={[]} />,
      { state: { engine: { backgroundState } } },
    );

    const disconnectButton = getByTestId(
      ConnectedAccountsSelectorsIDs.DISCONNECT,
    );
    fireEvent.press(disconnectButton);

    expect(mockNavigate).toHaveBeenCalledWith('Modal.RootModal', {
      screen: 'Sheet.RevokeAllAccountPermissions',
      params: expect.any(Object),
    });
  });

  it('handles add account button press', () => {
    const { getByTestId } = renderWithProvider(
      <AccountConnectMultiSelector {...defaultProps} />,
      { state: { engine: { backgroundState } } },
    );

    const addButton = getByTestId(
      AccountListBottomSheetSelectorsIDs.ACCOUNT_LIST_ADD_BUTTON_ID,
    );
    fireEvent.press(addButton);

    // Verify that the screen changes to AddAccountActions
    expect(mockNavigate).not.toHaveBeenCalled(); // Since this is handled internally
  });

  it('handles primary action button press when provided', () => {
    const { getByTestId } = renderWithProvider(
      <AccountConnectMultiSelector
        {...defaultProps}
        onPrimaryActionButtonPress={mockOnPrimaryActionButtonPress}
      />,
      { state: { engine: { backgroundState } } },
    );

    const updateButton = getByTestId(
      ConnectAccountBottomSheetSelectorsIDs.SELECT_MULTI_BUTTON,
    );
    fireEvent.press(updateButton);

    expect(mockOnPrimaryActionButtonPress).toHaveBeenCalled();
    expect(mockOnUserAction).not.toHaveBeenCalled();
  });

  xit('handles back button press', () => {
    const { getByText } = renderWithProvider(
      <AccountConnectMultiSelector {...defaultProps} />,
      { state: { engine: { backgroundState } } },
    );

    // Assuming SheetHeader renders a back button with text "back"
    const backButton = getByText('back');
    fireEvent.press(backButton);

    expect(mockOnBack).toHaveBeenCalled();
  });

  it('disables update button when no changes are made', () => {
    const { getByTestId } = renderWithProvider(
      <AccountConnectMultiSelector {...defaultProps} />,
      { state: { engine: { backgroundState } } },
    );

    const updateButton = getByTestId(
      ConnectAccountBottomSheetSelectorsIDs.SELECT_MULTI_BUTTON,
    );

    expect(updateButton.props.disabled).toBe(true);
  });
});
