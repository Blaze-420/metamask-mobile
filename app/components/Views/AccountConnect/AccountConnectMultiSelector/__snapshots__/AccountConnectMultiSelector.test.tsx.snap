// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountConnectMultiSelector renders correctly 1`] = `
<RCTSafeAreaView>
  <View
    style={
      {
        "height": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "flexDirection": "row",
          "height": 32,
          "justifyContent": "space-between",
          "margin": 16,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <TouchableOpacity
          accessible={true}
          activeOpacity={1}
          disabled={false}
          onPress={[MockFunction]}
          onPressIn={[Function]}
          onPressOut={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 8,
              "height": 24,
              "justifyContent": "center",
              "opacity": 1,
              "width": 24,
            }
          }
          testID="sheet-header-back-button"
        >
          <SvgMock
            color="#121314"
            height={16}
            name="ArrowLeft"
            style={
              {
                "height": 16,
                "width": 16,
              }
            }
            width={16}
          />
        </TouchableOpacity>
      </View>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#121314",
            "fontFamily": "CentraNo1-Bold",
            "fontSize": 18,
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 24,
          }
        }
      >
        Connect accounts
      </Text>
      <View
        style={
          {
            "alignItems": "flex-end",
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        {
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#686e7d",
            "fontFamily": "CentraNo1-Book",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 24,
            "marginVertical": 8,
            "textAlign": "center",
          }
        }
      >
        Select the account(s) to use on this site:
      </Text>
      <View>
        <TouchableOpacity
          disabled={false}
          onPress={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 24,
              "marginLeft": 0,
              "marginVertical": 12,
              "opacity": 1,
            }
          }
        >
          <View
            accessibilityRole="checkbox"
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#4459ff",
                "borderColor": "#4459ff",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 20,
                "justifyContent": "center",
                "width": 20,
              }
            }
          >
            <SvgMock
              color="#ffffff"
              height={20}
              name="MinusBold"
              onPress={[Function]}
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
              testID="checkbox-icon-component"
              width={20}
            />
          </View>
          <View
            style={
              {
                "marginLeft": 12,
              }
            }
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#121314",
                  "fontFamily": "CentraNo1-Book",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 24,
                }
              }
            >
              Select all
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
    <RCTScrollView
      collapsable={false}
      data={
        [
          {
            "address": "0x1234",
            "balance": "0x1",
            "isSelected": false,
            "name": "Account 1",
            "type": "Simple Key Pair",
            "yOffset": 0,
          },
          {
            "address": "0x5678",
            "balance": "0x2",
            "isSelected": false,
            "name": "Account 2",
            "type": "Simple Key Pair",
            "yOffset": 0,
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      initialNumToRender={999}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      testID="account-list"
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <TouchableOpacity
            buttonProps={
              {
                "buttonTestId": "main-wallet-account-actions-0",
                "onButtonClick": [Function],
              }
            }
            onLongPress={[Function]}
            onPress={[Function]}
            showSecondaryTextIcon={false}
            style={
              {
                "backgroundColor": "#ffffff",
                "borderRadius": 4,
                "opacity": 1,
                "padding": 16,
              }
            }
            testID="cellmultiselect"
          >
            <View
              accessibilityRole="none"
              accessible={true}
              style={
                {
                  "padding": 0,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <TouchableOpacity
                  disabled={false}
                  onPressIn={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 24,
                      "marginRight": -8,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    accessibilityRole="checkbox"
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#4459ff",
                        "borderColor": "#4459ff",
                        "borderRadius": 4,
                        "borderWidth": 2,
                        "height": 20,
                        "justifyContent": "center",
                        "width": 20,
                      }
                    }
                  >
                    <SvgMock
                      color="#ffffff"
                      height={20}
                      name="CheckBold"
                      onPressIn={[Function]}
                      style={
                        {
                          "height": 20,
                          "width": 20,
                        }
                      }
                      testID="checkbox-icon-component"
                      width={20}
                    />
                  </View>
                </TouchableOpacity>
                <View
                  accessible={false}
                  style={
                    {
                      "width": 16,
                    }
                  }
                  testID="listitem-gap"
                />
                <View
                  style={
                    {
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      {
                        "backgroundColor": "#ffffff",
                        "borderRadius": 16,
                        "height": 32,
                        "marginRight": 16,
                        "overflow": "hidden",
                        "width": 32,
                      }
                    }
                    testID="cellbase-avatar"
                  >
                    <View
                      style={
                        [
                          {
                            "overflow": "hidden",
                          },
                          {
                            "backgroundColor": "#F72401",
                            "borderRadius": 16,
                            "height": 32,
                            "width": 32,
                          },
                          undefined,
                        ]
                      }
                    >
                      <RNSVGSvgView
                        bbHeight={32}
                        bbWidth={32}
                        focusable={false}
                        height={32}
                        style={
                          [
                            {
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                            },
                            {
                              "flex": 0,
                              "height": 32,
                              "width": 32,
                            },
                          ]
                        }
                        width={32}
                      >
                        <RNSVGGroup
                          fill={
                            {
                              "payload": 4278190080,
                              "type": 0,
                            }
                          }
                        >
                          <RNSVGRect
                            fill={
                              {
                                "payload": 4294018050,
                                "type": 0,
                              }
                            }
                            height={32}
                            matrix={
                              [
                                -0.2923717047227371,
                                -0.9563047559630353,
                                0.9563047559630353,
                                -0.2923717047227371,
                                3.9314124245491167,
                                33.91803595917714,
                              ]
                            }
                            propList={
                              [
                                "fill",
                              ]
                            }
                            width={32}
                            x={0}
                            y={0}
                          />
                          <RNSVGRect
                            fill={
                              {
                                "payload": 4294157824,
                                "type": 0,
                              }
                            }
                            height={32}
                            matrix={
                              [
                                -0.21132479645538865,
                                0.9774158942860959,
                                -0.9774158942860959,
                                -0.21132479645538865,
                                42.603154834391255,
                                19.31548277122848,
                              ]
                            }
                            propList={
                              [
                                "fill",
                              ]
                            }
                            width={32}
                            x={0}
                            y={0}
                          />
                          <RNSVGRect
                            fill={
                              {
                                "payload": 4294727936,
                                "type": 0,
                              }
                            }
                            height={32}
                            matrix={
                              [
                                0.9887563810470058,
                                0.14953534344370972,
                                -0.14953534344370972,
                                0.9887563810470058,
                                -11.4311706408563,
                                -29.44153079403052,
                              ]
                            }
                            propList={
                              [
                                "fill",
                              ]
                            }
                            width={32}
                            x={0}
                            y={0}
                          />
                        </RNSVGGroup>
                      </RNSVGSvgView>
                    </View>
                  </View>
                  <View
                    style={
                      {
                        "alignItems": "flex-start",
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      accessibilityRole="text"
                      numberOfLines={1}
                      style={
                        {
                          "color": "#121314",
                          "fontFamily": "CentraNo1-Book",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 24,
                        }
                      }
                      testID="cellbase-avatar-title"
                    >
                      test1.eth
                    </Text>
                    <Text
                      accessibilityRole="text"
                      numberOfLines={1}
                      style={
                        {
                          "color": "#686e7d",
                          "fontFamily": "CentraNo1-Book",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 24,
                        }
                      }
                    >
                      0x1234...x1234
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessibilityRole="checkbox"
              accessible={true}
              style={
                {
                  "backgroundColor": "#4459ff1a",
                  "bottom": 0,
                  "flexDirection": "row",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
          </TouchableOpacity>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <TouchableOpacity
            buttonProps={
              {
                "buttonTestId": "main-wallet-account-actions-1",
                "onButtonClick": [Function],
              }
            }
            onLongPress={[Function]}
            onPress={[Function]}
            showSecondaryTextIcon={false}
            style={
              {
                "backgroundColor": "#ffffff",
                "borderRadius": 4,
                "opacity": 1,
                "padding": 16,
              }
            }
            testID="cellmultiselect"
          >
            <View
              accessibilityRole="none"
              accessible={true}
              style={
                {
                  "padding": 0,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <TouchableOpacity
                  disabled={false}
                  onPressIn={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 24,
                      "marginRight": -8,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    accessibilityRole="checkbox"
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#ffffff",
                        "borderColor": "#121314",
                        "borderRadius": 4,
                        "borderWidth": 2,
                        "height": 20,
                        "justifyContent": "center",
                        "width": 20,
                      }
                    }
                  />
                </TouchableOpacity>
                <View
                  accessible={false}
                  style={
                    {
                      "width": 16,
                    }
                  }
                  testID="listitem-gap"
                />
                <View
                  style={
                    {
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      {
                        "backgroundColor": "#ffffff",
                        "borderRadius": 16,
                        "height": 32,
                        "marginRight": 16,
                        "overflow": "hidden",
                        "width": 32,
                      }
                    }
                    testID="cellbase-avatar"
                  >
                    <View
                      style={
                        [
                          {
                            "overflow": "hidden",
                          },
                          {
                            "backgroundColor": "#017B8C",
                            "borderRadius": 16,
                            "height": 32,
                            "width": 32,
                          },
                          undefined,
                        ]
                      }
                    >
                      <RNSVGSvgView
                        bbHeight={32}
                        bbWidth={32}
                        focusable={false}
                        height={32}
                        style={
                          [
                            {
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                            },
                            {
                              "flex": 0,
                              "height": 32,
                              "width": 32,
                            },
                          ]
                        }
                        width={32}
                      >
                        <RNSVGGroup
                          fill={
                            {
                              "payload": 4278190080,
                              "type": 0,
                            }
                          }
                        >
                          <RNSVGRect
                            fill={
                              {
                                "payload": 4294740992,
                                "type": 0,
                              }
                            }
                            height={32}
                            matrix={
                              [
                                0.22835087011065588,
                                0.9735789028731602,
                                -0.9735789028731602,
                                0.22835087011065588,
                                31.410188562955305,
                                -2.5909613952202437,
                              ]
                            }
                            propList={
                              [
                                "fill",
                              ]
                            }
                            width={32}
                            x={0}
                            y={0}
                          />
                          <RNSVGRect
                            fill={
                              {
                                "payload": 4291236924,
                                "type": 0,
                              }
                            }
                            height={32}
                            matrix={
                              [
                                -0.12186934340514737,
                                0.9925461516413221,
                                -0.9925461516413221,
                                -0.12186934340514737,
                                44.29656540917626,
                                5.665580734505106,
                              ]
                            }
                            propList={
                              [
                                "fill",
                              ]
                            }
                            width={32}
                            x={0}
                            y={0}
                          />
                          <RNSVGRect
                            fill={
                              {
                                "payload": 4294170368,
                                "type": 0,
                              }
                            }
                            height={32}
                            matrix={
                              [
                                -0.978867388761685,
                                -0.2044960518417903,
                                0.2044960518417903,
                                -0.978867388761685,
                                42.76161861053511,
                                56.268492097304716,
                              ]
                            }
                            propList={
                              [
                                "fill",
                              ]
                            }
                            width={32}
                            x={0}
                            y={0}
                          />
                        </RNSVGGroup>
                      </RNSVGSvgView>
                    </View>
                  </View>
                  <View
                    style={
                      {
                        "alignItems": "flex-start",
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      accessibilityRole="text"
                      numberOfLines={1}
                      style={
                        {
                          "color": "#121314",
                          "fontFamily": "CentraNo1-Book",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 24,
                        }
                      }
                      testID="cellbase-avatar-title"
                    >
                      test2.eth
                    </Text>
                    <Text
                      accessibilityRole="text"
                      numberOfLines={1}
                      style={
                        {
                          "color": "#686e7d",
                          "fontFamily": "CentraNo1-Book",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 24,
                        }
                      }
                    >
                      0x5678...x5678
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "marginBottom": 16,
          "marginHorizontal": 16,
        }
      }
    >
      <TouchableOpacity
        accessibilityRole="button"
        accessible={true}
        activeOpacity={1}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "transparent",
            "borderRadius": 24,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "center",
            "paddingHorizontal": 16,
          }
        }
        testID="account-list-add-account-button"
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#4459ff",
              "fontFamily": "CentraNo1-Medium",
              "fontSize": 16,
              "fontWeight": "500",
              "letterSpacing": 0,
              "lineHeight": 24,
            }
          }
        >
          Add account or hardware wallet
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        {
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        style={
          {
            "marginBottom": 0,
            "marginTop": 0,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={1}
            disabled={true}
            onPress={[Function]}
            onPressIn={[Function]}
            onPressOut={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "#4459ff",
                "borderRadius": 24,
                "flex": 1,
                "flexDirection": "row",
                "height": 48,
                "justifyContent": "center",
                "opacity": 0.5,
                "paddingHorizontal": 16,
              }
            }
            testID="multiconnect-connect-button"
          >
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#ffffff",
                  "fontFamily": "CentraNo1-Medium",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0,
                  "lineHeight": 24,
                }
              }
            >
              Update
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
